cmd/oras/pull.go

```diff
@@ -61,106 +66,60 @@ Example - Pull files with local cache:
-	cmd.Flags().BoolVarP(&opts.pathTraversal, "allow-path-traversal", "T", false, "allow storing files out of the output directory")
-	cmd.Flags().StringVarP(&opts.output, "output", "o", "", "output directory")
-	cmd.Flags().StringVarP(&opts.manifestConfigRef, "manifest-config", "", "", "output manifest config file")
-	cmd.Flags().BoolVarP(&opts.verbose, "verbose", "v", false, "verbose output")
-
-	cmd.Flags().BoolVarP(&opts.debug, "debug", "d", false, "debug mode")
-	cmd.Flags().StringArrayVarP(&opts.configs, "config", "c", nil, "auth config path")
-	cmd.Flags().StringVarP(&opts.username, "username", "u", "", "registry username")
-	cmd.Flags().StringVarP(&opts.password, "password", "p", "", "registry password")
-	cmd.Flags().BoolVarP(&opts.insecure, "insecure", "", false, "allow connections to SSL registry without certs")
-	cmd.Flags().BoolVarP(&opts.plainHTTP, "plain-http", "", false, "use plain http and not https")
+	cmd.Flags().BoolVarP(&opts.KeepOldFiles, "keep-old-files", "k", false, "do not replace existing files when pulling, treat them as errors")
+	cmd.Flags().BoolVarP(&opts.PathTraversal, "allow-path-traversal", "T", false, "allow storing files out of the output directory")
+	cmd.Flags().StringVarP(&opts.Output, "output", "o", ".", "output directory")
+	cmd.Flags().StringVarP(&opts.ManifestConfigRef, "manifest-config", "", "", "output manifest config file")
+	option.ApplyFlags(&opts, cmd.Flags())
 	return cmd
 }
 
 func runPull(opts pullOptions) error {
-	ctx := context.Background()
-	if opts.debug {
-		logrus.SetLevel(logrus.DebugLevel)
-	} else if !opts.verbose {
-		ctx = ctxo.WithLoggerDiscarded(ctx)
+	ctx, _ := opts.SetLoggerLevel()
+	repo, err := opts.NewRepository(opts.targetRef, opts.Common)
+	if err != nil {
+		return err
 	}
-	resolver := newResolver(opts.username, opts.password, opts.insecure, opts.plainHTTP, opts.configs...)
-	store := content.NewFileStore(opts.output)
-	defer store.Close()
-	store.DisableOverwrite = opts.keepOldFiles
-	store.AllowPathTraversalOnWrite = opts.pathTraversal
-
-	pullOpts := []oras.PullOpt{
-		oras.WithPullStatusTrack(os.Stdout),
+	var dstStore = file.New(opts.Output)
+	dstStore.AllowPathTraversalOnWrite = opts.PathTraversal
+	dstStore.DisableOverwrite = opts.KeepOldFiles
+	var mco *status.ManifestConfigOption
+	if opts.ManifestConfigRef != "" {
+		name, media := parseFileRef(opts.ManifestConfigRef, oras.MediaTypeUnknownConfig)
+		mco = &status.ManifestConfigOption{
+			Name:      name,
+			MediaType: media,
+		}
 	}
+	var cache oras.Target
 	if opts.cacheRoot != "" {
-		cachedStore, err := newStoreWithCache(store, opts.cacheRoot)
+		cache, err = oci.New(opts.cacheRoot)
 		if err != nil {
 			return err
 		}
-		pullOpts = append(pullOpts, oras.WithContentProvideIngester(cachedStore))
-	}
-	if opts.manifestConfigRef != "" {
-		pullOpts = appendPullManifestConfigHandlers(pullOpts, opts.manifestConfigRef)
 	}
+	tracker := status.NewPullTracker(dstStore, mco, cache)
```

Comment:
The cache implementation should be decoupled with the status tracker.

A suggested way of implementation is to create a cache of type  `oras.Target` by wrapping the destination `oras.Target`.